src/main/docs/guide/azureCosmos/azureCosmosOptimisticLocking.adoc

```diff
@@ -0,0 +1,7 @@
+Optimistic locking is a strategy where you note the actual record state's version and modify the record only when the version is the same.
+
+Unlike some other Db implementations in Micronaut, for Azure Cosmos we rely on existence of _etag field in every document. We don't use ann:data.annotation.Version[] because _etag field is of type `String` and for that purpose we introduce ann:io.micronaut.data.cosmos.annotation.ETag[] annotation.
```

Comment:
I think we can support `@Version` to use `_etag` in the future. Are there any examples of how can I use this etag for some pessimistic locking?

Reply:
There is Family class having `@ETag` annotation mapped to `String documentVersion` field below in this section.

Reply:
How is it usefull? What is the use case? Can we use it for the pessimistic locking?

Reply:
Optimistic only, works similar to `@Version` if I am not wrong. That internal _etag field is updated each time document is updated. There is request option `setIfMatchETag` to set the value of the document before being updated. If some other user updated document in the meantime, Azure Cosmos Db compares _etag version and returns 412 status so we can know when to throw `OptimisticLockException`.